akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/common/ElasticDAO.scala

```diff
@@ -0,0 +1,174 @@
+                            )
+
+  case class DataSavedConfirmation(index: String, typeName: String, jsonDocs: Seq[ElasticDocumentInsertable])
+
+  case class DeleteFromIndex(
+                              index: String,
+                              typeName: String,
+                              where: Option[QueryDefinition] = None
+                            )
+
+  case class ErrorDeletingData(
+                              error: Exception,
+                              index: String,
+                              typeName: String,
+                              originalRQ: DeleteFromIndex
+                              )
+
+  case class DeleteConfirmation(index: String, typeName: String, originalRQ: DeleteFromIndex)
+
+  case class RetrieveAllFromIndexSorted(
+                                         index: String,
+                                         typeName: String,
+                                         where: Option[QueryDefinition] = None,
+                                         sort: Option[FieldSortDefinition] = None,
+                                         limit: Option[Int] = None)
+
+  case class DataRetrieved(
+                            index: String,
+                            typeName: String,
+                            data: RichSearchResponse,
+                            originalRQ: RetrieveAllFromIndexSorted
+                          )
+
+  case class ErrorRetrievingData(
+                                  error: Exception,
+                                  index: String,
+                                  typeName: String,
+                                  originalRQ: RetrieveAllFromIndexSorted
+                                )
+
+  object InitElasticDAO
+  case class ProblemConnectingWithES(e: Exception)
+  object ReadyToListen
+
+  private val settings = Settings.builder()
+    .put("cluster.name", TapiporlaConfig.ElasticSearch.clusterName)
+    .put("xpack.security.user", TapiporlaConfig.ElasticSearch.authUser)
+    .build()
+
+  val client = XPackElasticClient(settings, ElasticsearchClientUri(TapiporlaConfig.ElasticSearch.endpoint))
+
+}
+
+
+trait ElasticDAO extends Actor with ActorLogging with Stash{
+
+  def indexCreation: CreateIndexDefinition //Index to be created (if already exists we don't check if the mappings are similar)
+
+  self ! InitElasticDAO
+
+  override def receive = initial
+
+  def initial: Receive = {
+
+    case InitElasticDAO =>
+      client.execute(indexCreation) map { _ =>
+        ReadyToListen
+      } recover {
+        case CausedBy(ex: ResourceAlreadyExistsException) => //TODO: Change to see how to check if index already exists
+          ReadyToListen
+        case e: Exception =>
+          ProblemConnectingWithES(e)
+
+      } pipeTo (self)
+
+
+    case ProblemConnectingWithES(e) =>
+      log.error(s"Error initializing ElasticSearch DAO ${self.path.name} :$e. Trying again in 30 seconds")
```

Comment:
Are we sure that ```self.path.name``` display can't be configured via logger pattern?

Reply:
Self.path.name was the worst property I used to log ever. But I don't know I saw it, and I tried it without removing them. XD Removed :dagger: